java/client/src/org/openqa/selenium/DeviceRotation.java

```diff
@@ -54,14 +54,14 @@ public DeviceRotation(int x, int y, int z) {
 	 * z : zVal
 	 * @param map
 	 */
-	public DeviceRotation(Map<String, Integer> map) {
+	public DeviceRotation(Map<String, Long> map) {
 		if (map == null || !map.containsKey("x") || !map.containsKey("y") || !map.containsKey("z")) {
 			throw new IllegalArgumentException("Could not initialize DeviceRotation with map given: " + map.toString());
 		}
-		this.validateParameters(map.get("x"), map.get("y"), map.get("z"));
-		this.x = map.get("x");
-		this.y = map.get("y");
-		this.z = map.get("z");
+		this.validateParameters(map.get("x").intValue(), map.get("y").intValue(), map.get("z").intValue());
```

Comment:
put the validate line at the end and then you can do just:

    this.validateParameters(x, y, z);

also, in validateParameters can you adjust it to ensure the values are less than 360 too (as per spec)